server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesRequest.java

```diff
@@ -117,6 +128,24 @@ public FieldCapabilitiesRequest fields(String... fields) {
         return fields;
     }
 
+    public FieldCapabilitiesRequest filters(String... filters) {
+        this.filters = filters;
+        return this;
+    }
+
+    public String[] filters() {
+        return filters;
+    }
+
+    public FieldCapabilitiesRequest allowedTypes(String... types) {
```

Comment:
nit: why call this allowedTypes compared to just types?

Reply:
I changed this a few times - just `types` doesn't really get across the idea that this is an *inclusive* filter, as opposed to the other filters.  But that applies to the field names as well, I suppose.  I can change it.

Reply:
I think it would be nice to have the names aligned with their names on the rest layer, it's easier to follow where things go when parsed to the request object